pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java

```diff
@@ -38,22 +41,24 @@
  */
 @Slf4j
 public class JavaInstance implements AutoCloseable {
+    private final Logger instanceLog;
+    private ContextImpl context;
 
     @Getter(AccessLevel.PACKAGE)
-    private final ContextImpl context;
     private Function function;
     private java.util.function.Function javaUtilFunction;
+    private Optional<PulsarSource> optionalPulsarSource = Optional.empty();
 
     public JavaInstance(InstanceConfig config, Object userClassObject,
                  ClassLoader clsLoader,
                  PulsarClient pulsarClient,
                  Source source) {
         // TODO: cache logger instances by functions?
-        Logger instanceLog = LoggerFactory.getLogger("function-" + config.getFunctionDetails().getName());
+        this.instanceLog = LoggerFactory.getLogger("function-" + config.getFunctionDetails().getName());
```

Comment:
I don't think we need a instanceLog to be a member variable.  I think it is only referenced in the constructor